src/node_credentials.cc

```diff
@@ -33,11 +37,38 @@ bool linux_at_secure = false;
 
 namespace credentials {
 
-// Look up environment variable unless running as setuid root.
+#if defined(__linux__)
+// Returns true if the current process only has the passed-in capability.
+bool HasOnly(int capability) {
+  DCHECK(cap_valid(capability));
+
+  struct __user_cap_data_struct cap_data[2];
+  struct __user_cap_header_struct cap_header_data = {
+    _LINUX_CAPABILITY_VERSION_3,
+    getpid()};
+
+
+  if (syscall(SYS_capget, &cap_header_data, &cap_data[0]) != 0) {
+    return false;
+  }
+  return cap_data[0].effective ==
```

Comment:
We should check the `permitted` set, not `effective`, no? What about `cap_data[1]`?

Reply:
> We should check the permitted set, not effective, no?

Yes, we should. Sorry about that, I've been fooling myself while testing and setting the capabilities to `+ep` instead of just `+p` and did not notice that the code is checking the effective and not the permitted set. Thanks!

>What about cap_data[1]?

Yeah, we should be checking that. I'll update shortly. Thanks again.